gnd/src/main/java/com/google/android/gnd/repository/ProjectRepository.java

```diff
@@ -166,4 +173,20 @@ public void clearActiveProject() {
     localValueStore.clearLastActiveProjectId();
     selectProjectEvent.onNext(Optional.empty());
   }
+
+  public ImmutableList<Layer> getModifiableLayers(Project project, User user) {
+    String role = project.getAcl().get(user.getEmail());
+    if (role == null) {
+      return ImmutableList.of();
+    }
+    return stream(project.getLayers())
+        .filter(layer -> canAddFeatures(role, layer))
+        .collect(toImmutableList());
+  }
+
+  private boolean canAddFeatures(String role, Layer layer) {
+    return OWNER.equals(role)
+        || MANAGER.equals(role)
+        || CONTRIBUTOR.equals(role) && !layer.getContributorsCanAdd().isEmpty();
```

Comment:
Shouldn't this check for membership in the layer's contributors "can add" list?

Reply:
The can add list contains what types of features the user can add, as opposed to the user id. This basically says that users can add features to a layer if they're owners, managers, or if they're contributors and they layer allows contributors to add any type of feature ('point' or 'polygon').